### PR TITLE
Homework12

### PR DESCRIPTION
**Pull request by LilianaLukash**

### CODE REVIEW

File: 12/function_calls.txt [hunk 1]
```diff
@@ -0,0 +1,12 @@
+my_function called at 1673820764.4037929
```
Comment (acman) [inline 12/function_calls.txt]: better not add the environmental and test files to commit like .idea, .txt, only code

File: 12/hw.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
+def mydecorator(func):
```
Comment (acman) [inline 12/hw.py]: better name decorator about how it works like `log_decorator`, `log_execution_time_decorator`.

File: 12/hw.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+        print(start)
+        stname = func.__name__
+        print(stname)
+        return_value = func(*args, **kwargs)
+        return return_value
+    return wrapper
+
+@mydecorator
+def littlefunction():
+    print("Ola amigos!")
```
Comment (acman) [inline 12/hw.py:15]: Hola! :)

File: 12/hw13a.py [hunk 4]
```diff
@@ -0,0 +1,54 @@
+import json
+phonebook = {"lili":"5555137"}
+
+commands = ["stats", "add", "delete", "list", "show", "exit"]
+while True:
+    com = (input("Please choose your command: stats, add, delete <name>, list, show <name>, exit ")).split()
+    with open("phonebook.txt", "r") as book:
```
Comment (acman) [inline 12/hw13a.py]: `phonebook.json`

File: 12/hw13c.py [hunk 5]
```diff
@@ -0,0 +1,9 @@
+import time
+class MyCustomException(Exception):
+    def __init__(self):
+        self.message = "Custom exception is occured"
+        with open("myexeption.txt", "a") as f:
+            print(f"{self.__class__.__name__} at {time.time()}", file=f)
+        super().__init__(self.message)
+
+raise MyCustomException()
```
Comment (acman) [inline 12/hw13c.py]: better to add empty line at the end of all python files

**Review by acman — CHANGES_REQUESTED** [review]
LGTM! there are only small fixes that should be provided

File: 12/hw.py (hunk 2 above)
Comment (LilianaLukash) [reply 12/hw.py]: Thanks, Corrected

File: 12/hw.py (hunk 3 above)
Comment (LilianaLukash) [reply 12/hw.py:15]: I am in Portugal and have to learn Portuguise) So here its Ola))

File: 12/hw13a.py (hunk 4 above)
Comment (LilianaLukash) [reply 12/hw13a.py]: corrected in new commit

File: 12/hw13c.py (hunk 5 above)
Comment (LilianaLukash) [reply 12/hw13c.py]: corrected

**Review by LilianaLukash — COMMENTED** [review]
corrected

**Review by acman — APPROVED** [review]
nice job! but next time dont add `.idea/` directory to commit or create .gitignore for it

**Pull request merged**